vscode-swift-language-server/src/swiftSourceTypes.ts

```diff
@@ -1,306 +1,291 @@
-    Swift |
-    Decl |
-    DeclExtension |
-    DeclFunction |
-    DeclFunctionOperator |
-    DeclFunctionAccessor |
-    DeclFunctionMethod |
-    DeclVar |
-    Ref |
-    RefFunction |
-    RefFunctionMethod |
-    RefFunctionOperator |
-    RefVar |
-    Syntaxtype |
-    SyntaxtypeAttribute;
+    let Keyword: 'source.lang.swift.keyword';
+    let DeclAssociatedtype: 'source.lang.swift.decl.associatedtype';
+    let DeclClass: 'source.lang.swift.decl.class';
+    let DeclEnum: 'source.lang.swift.decl.enum';
+    let DeclEnumelement: 'source.lang.swift.decl.enumelement';
+    let DeclGenericTypeParam: 'source.lang.swift.decl.generic_type_param';
+    let DeclProtocol: 'source.lang.swift.decl.protocol';
 
-/**
- * Maps SourceKit types to VSCode {SymbolKind}
- * FIXME: kinda limted
- *
- * @export
- * @param {SwiftType} swiftType
- * @returns {SymbolKind}
- */
-export function symbolKindForSwiftType(swiftType: SwiftType): SymbolKind {
-    // not really easy to have an obvious completion kind per type.
-    switch (swiftType) {
-// high confidence
-        case Decl.Module:
-            return SymbolKind.Module;
-        case Decl.Enum:
-        case Decl.Enumelement:
-        case Ref.Enum:
-        case Ref.Enumelement:
-            return SymbolKind.Enum;
-// medium confidence
-        case DeclFunction.Free:
-        case DeclFunction.Subscript:
-        case DeclFunctionOperator.Infix:
-        case RefFunction.Free:
-        case RefFunction.Subscript:
-        case RefFunctionOperator.Infix:
-            return SymbolKind.Function;
-        case DeclFunctionMethod.Class:
-        case DeclFunctionMethod.Static:
-        case DeclFunctionMethod.Instance:
-            return SymbolKind.Method;
-        case DeclFunction.Constructor:
-        case RefFunction.Constructor:
-            return SymbolKind.Constructor;
-        case DeclFunctionAccessor.Getter:
-        case DeclFunctionAccessor.Setter:
-            return SymbolKind.Field;
-        case Decl.Associatedtype:
-        case Decl.Class:
-        case Decl.Struct:
-        case DeclExtension.Class:
-        case Ref.Class:
-        case Ref.Struct:
-            return SymbolKind.Class;
-        case RefFunctionMethod.Class:
-        case RefFunctionMethod.Instance:
-            return SymbolKind.Method;
-// low confidence
-        case DeclVar.Global:
-        case DeclVar.Instance:
-        case DeclVar.Local:
-        case RefVar.Global:
-        case RefVar.Instance:
-        case RefVar.Local:
-            return SymbolKind.Variable;
-        case Syntaxtype.Argument:
-        case Syntaxtype.Parameter:
-            return SymbolKind.Field;
-        case Ref.Associatedtype:
-        case Decl.GenericTypeParam:
-        case Ref.GenericTypeParam:
-        case Ref.Protocol:
-        case Ref.Typealias:
-        case Decl.Protocol:
-        case Decl.Typealias:
-            return SymbolKind.Interface;
+    let DeclStruct: 'source.lang.swift.decl.struct';
```

Comment:
These seem like constants to me. Should they be `const` instead of `let` (or does TypeScript handle that transparently)?

Forgive me if that question was impertinent, I'm still new to TypeScript so I'm not always sure were the magic starts and ends.